### PR TITLE
Adds flat world IMU publisher for TurtleBot 2.

### PR DESCRIPTION
**Pull request by damonkohler**

### CODE REVIEW

File: cartographer_turtlebot2/src/flat_world_imu_node_main.cc [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  ros::Time last_time;
+  ::ros::NodeHandle node_handle("~");
+  ::ros::Publisher publisher =
+      node_handle.advertise<sensor_msgs::Imu>(kFlatWorldImuTopic, 10);
+  ::ros::Subscriber subscriber = node_handle.subscribe(
+      kImuTopic, kSubscriberQueueSize,
+      boost::function<void(const sensor_msgs::Imu::ConstPtr& msg)>(
+          [&](const sensor_msgs::Imu::ConstPtr& msg) {
+            if (!last_time.isZero()) {
```
Comment (wohe) [inline cartographer_turtlebot2/src/flat_world_imu_node_main.cc]: Why not publish in this case? How about changing it to
if (last_time.isZero() || msg->header.stamp > last_time) {
?

File: cartographer_turtlebot2/src/flat_world_imu_node_main.cc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+              if (msg->header.stamp > last_time) {
+                sensor_msgs::Imu flat_world_imu = *msg;
+                flat_world_imu.linear_acceleration.x = 0.;
+                flat_world_imu.linear_acceleration.y = 0.;
+                flat_world_imu.linear_acceleration.z = 10.;
```
Comment (wohe) [inline cartographer_turtlebot2/src/flat_world_imu_node_main.cc]: 9.8? And maybe define a kFakeGravity constant?

File: cartographer_turtlebot2/src/flat_world_imu_node_main.cc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+              }
+            }
+            last_time = msg->header.stamp;
+          }));
+
+  ::ros::start();
+  ::ros::spin();
+  ::ros::shutdown();
+
+  return 0;
```
Comment (wohe) [inline cartographer_turtlebot2/src/flat_world_imu_node_main.cc]: Unnecessary. Remove?

File: cartographer_turtlebot2/package.xml [hunk 4]
```diff
@@ -34,6 +34,9 @@
   <depend>depthimage_to_laserscan</depend>
   <depend>kobuki</depend>
   <depend>kobuki_core</depend>
+  <depend>message_runtime</depend>
   <depend>openni2_launch</depend>
+  <depend>roscpp</depend>
```
Comment (wohe) [inline cartographer_turtlebot2/package.xml]: Should we also add <build_depend>g++-static</build_depend> as in cartographer_ros?

File: cartographer_turtlebot2/configuration_files/turtlebot2.lua [hunk 5]
```diff
@@ -28,16 +28,17 @@ options = {
   use_horizontal_laser = true,
   use_horizontal_multi_echo_laser = false,
   horizontal_laser_min_range = 0.,
-  horizontal_laser_max_range = 30.,
-  horizontal_laser_missing_echo_ray_length = 5.,
+  horizontal_laser_max_range = 4.,
```
Comment (wohe) [inline cartographer_turtlebot2/configuration_files/turtlebot2.lua]: Revert? Doesn't this mess up the LIDAR support?

File: cartographer_turtlebot2/launch/demo_turtlebot2.launch [hunk 6]
```diff
@@ -22,7 +22,13 @@
           -configuration_directory
               $(find cartographer_turtlebot2)/configuration_files
           -configuration_basename turtlebot2.lua"
-      output="screen" />
+      output="screen">
+      <remap from="imu" to="/flat_world_imu" />
+  </node>
+  <node name="flat_world_imu" pkg="cartographer_turtlebot2"
```
Comment (wohe) [inline cartographer_turtlebot2/launch/demo_turtlebot2.launch]: I think this needs to be added to turtlebot2.launch as well.

File: cartographer_turtlebot2/package.xml (hunk 4 above)
Comment (damonkohler) [inline cartographer_turtlebot2/package.xml]: Done.

File: cartographer_turtlebot2/src/flat_world_imu_node_main.cc (hunk 1 above)
Comment (damonkohler) [inline cartographer_turtlebot2/src/flat_world_imu_node_main.cc]: Done.

File: cartographer_turtlebot2/src/flat_world_imu_node_main.cc (hunk 2 above)
Comment (damonkohler) [inline cartographer_turtlebot2/src/flat_world_imu_node_main.cc]: Done.

File: cartographer_turtlebot2/src/flat_world_imu_node_main.cc (hunk 3 above)
Comment (damonkohler) [inline cartographer_turtlebot2/src/flat_world_imu_node_main.cc]: Done.

File: cartographer_turtlebot2/package.xml (hunk 4 above)
Comment (damonkohler) [inline cartographer_turtlebot2/package.xml]: Done.

File: cartographer_turtlebot2/configuration_files/turtlebot2.lua (hunk 5 above)
Comment (damonkohler) [inline cartographer_turtlebot2/configuration_files/turtlebot2.lua]: Done.

File: cartographer_turtlebot2/launch/demo_turtlebot2.launch (hunk 6 above)
Comment (damonkohler) [inline cartographer_turtlebot2/launch/demo_turtlebot2.launch]: Done.

File: cartographer_turtlebot2/launch/demo_turtlebot2.launch [hunk 7]
```diff
@@ -22,7 +22,13 @@
           -configuration_directory
               $(find cartographer_turtlebot2)/configuration_files
           -configuration_basename turtlebot2.lua"
-      output="screen" />
+      output="screen">
+  </node>
```
Comment (wohe) [inline cartographer_turtlebot2/launch/demo_turtlebot2.launch]: nit: Revert to " />"?

File: cartographer_turtlebot2/launch/demo_turtlebot2.launch [hunk 8]
```diff
@@ -22,7 +22,13 @@
...
+  <node name="flat_world_imu" pkg="cartographer_turtlebot2"
+      type="flat_world_imu_node" output="screen">
+      <remap from="~/imu" to="/mobile_base/sensors/imu_data_raw" />
```
Comment (wohe) [inline cartographer_turtlebot2/launch/demo_turtlebot2.launch]: Indent by 2 spaces only for nested tags?

File: cartographer_turtlebot2/src/flat_world_imu_node_main.cc [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+constexpr char kImuTopic[] = "imu";
+constexpr char kFlatWorldImuTopic[] = "flat_world_imu";
+
+}  // namespace
+
+int main(int argc, char** argv) {
+  ::ros::init(argc, argv, "flat_world_imu");
+
+  ros::Time last_time;
+  ::ros::NodeHandle node_handle("~");
```
Comment (wohe) [inline cartographer_turtlebot2/src/flat_world_imu_node_main.cc]: As discussed offline: Remove ~ here and elsewhere, and let users configure their namespaces.

Comment (damonkohler) [inline cartographer_turtlebot2/src/flat_world_imu_node_main.cc]: Done.

**Pull request merged**